libraries/plugins/xfpga/mmio.c

```diff
@@ -319,13 +319,21 @@ fpga_result __XFPGA_API__ xfpga_fpgaReadMMIO64(fpga_handle handle,
 	return result;
 }
 
+#if (defined(__i386__) || defined(__x86_64__) || defined(__ia64__)) && GCC_VERSION >= 40900
 static inline void copy512(const void *src, void *dst)
```

Comment:
I'm wondering if we should change the signature of this function to return either 0 on success and non-zero on failure.
Or maybe put some logging information to warn when the dummy version is called. I would like to have some sort of indication that calling that version is essentially a no-op.

Reply:
We'll get FPGA_NOT_SUPPORTED in the case of non-Intel architecture, before copy512() is actually called.